Split integrate.md to itself and ingress-integration.md (#713)
canonical/traefik-k8s-operator: docs/reference/ingress-integration.md

---
myst:
  html_meta:

>     "description lang=en": "Learn more about the ingress-related integrations of the Traefik charm."

Reviewer: ```suggestion
    "description lang=en": "Learn more about the ingress-related relations (integrations) of the Traefik charm."
```

---

(reference_ingress_integrations)=

> # Ingress-related integrations

Reviewer: ```suggestion
# Ingress-related relations
```

The [Traefik charm](https://charmhub.io/traefik-k8s) is a charm to provide ingress to
another charmed application 'the Juju way'. The idea is that if a charm integrates with
`traefik-k8s` then you can relate the two applications and your application will

> receive the URL at which ingress is made available.

Reviewer: ```suggestion
The [Traefik charm](https://charmhub.io/traefik-k8s) provides ingress to
another charmed application through a Juju relation. The idea is that if a charm integrates with
`traefik-k8s` then you can integrate the two applications and your application will
receive the URL at which ingress is made available.
```
nits on the phrasing

The Traefik charm supports two standardized interfaces:

> - [ingress](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/ingress/v2#readme) 

Reviewer: ```suggestion
- [`ingress`](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/ingress/v2#readme) 
```
nit, to be consistent with the other interface

  Using this interface, each charmed application can request a single, cluster-unique URL for ingress.
  You can choose between a domain-name-based URL (`your.parameters.domain.com`) and a path-based URL (`domain.com\your\parameters`).
- [`ingress-per-unit`](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/ingress_per_unit/v0/README.md)

  Using this interface, each charmed application can request a cluster-unique URL for each

>   existing unit. This is for applications such as Prometheus, where each remote-write

Reviewer: ```suggestion
  existing unit. This is ideal for applications such as Prometheus, where each remote-write
```
nit

  endpoint needs to be routed to separately, and database applications who wish to do
  client-side load-balancing.

## Traefik route charm

The [Traefik route charm](https://charmhub.io/traefik-route-k8s) is a proxy charm that sits
between Traefik and a charm in need of ingress, and is used to provide low-level access to
Traefik configuration, as well as to allow configuration for each relation. 

Want to have full access to all the expressive power of
[Traefik's routing configuration](https://doc.traefik.io/traefik/routing/overview/)? Want to
have one Traefik instance, and provide domain-name-based URL routing to some charms, but

> path-based URL routing to some others? This is how you do it.

Reviewer: This reads more like marketing material rather than a technical description, maybe we can rephrase to something like:
```suggestion
This charm is ideal for use cases where you need the full expressive power of
[Traefik's routing configuration](https://doc.traefik.io/traefik/routing/overview/),
or if you want to use a single Traefik instance to provide domain-name-based
URL routing to some charms, but path-based URL routing to others.
```
